subprojects/plugin-use/src/integTest/groovy/org/gradle/plugin/use/resolve/service/ScriptPluginUseIntegrationTest.groovy

```diff
@@ -0,0 +1,412 @@
+
+        and:
+        server.expectGet("/bye.gradle", bye)
+        server.start()
+
+        and:
+        buildFile << """
+
+            plugins {
+                script("gradle/hello.gradle")
+                script("http://localhost:${server.port}/bye.gradle")
+            }
+
+        """.stripIndent()
+
+        when:
+        succeeds "help"
+
+        then:
+        output.contains("Hello from the other side")
+        output.contains("Bye from the other side")
+
+        and:
+        operations.hasOperation("Apply script plugin 'gradle/hello.gradle' to root project 'root'")
+        operations.hasOperation("Apply script plugin 'http://localhost:${server.port}/bye.gradle' to root project 'root'")
+    }
+
+    def "project scripts in a multi-projects build can request script plugins"() {
+
+        given:
+        file("gradle/hello.gradle") << "println(\"Hello from the other side of \$name\")"
+
+        and:
+        settingsFile << """
+
+            rootProject.name = "root"
+            include "a", "b"
+
+        """.stripIndent()
+
+        and:
+        buildFile << """
+
+            plugins {
+                script("gradle/hello.gradle")
+            }
+
+        """.stripIndent()
+        file("a/build.gradle") << """
+
+            plugins {
+                script("../gradle/hello.gradle")
+            }
+
+        """.stripIndent()
+        file("b/build.gradle") << """
+
+            plugins {
+                script("../gradle/hello.gradle")
+            }
+
+        """.stripIndent()
+
+        when:
+        succeeds "help"
+
+        then:
+        output.contains("Hello from the other side of root")
+        output.contains("Hello from the other side of a")
+        output.contains("Hello from the other side of b")
+
+        and:
+        operations.hasOperation("Apply script plugin 'gradle/hello.gradle' to root project 'root'")
+        operations.hasOperation("Apply script plugin 'gradle/hello.gradle' to project ':a'")
+        operations.hasOperation("Apply script plugin 'gradle/hello.gradle' to project ':b'")
+    }
+
+    def "same script plugin requested from different paths gets applied only once"() {
+
+        given:
+        file("gradle/hello.gradle") << "println('Hello from the other side')"
+
+        and:
+        buildFile << """
+
+            plugins {
+                script("gradle/hello.gradle")
+                script("gradle/../gradle/hello.gradle")
+            }
+
+        """.stripIndent()
+
+        when:
+        succeeds "help", "-q"
+
+        then:
+        output.startsWith """
+            Hello from the other side
+            
+            Welcome to Gradle
+            """.stripIndent().trim()
+
+        and:
+        operations.all(ApplyPluginBuildOperationType) { it.displayName.contains("hello.gradle") }.size() == 1
+    }
+
+    def "same script plugin requested from different URLs gets applied once"() {
```

Comment:
I'm not sure about using the content of the script as its identity. I would probably expect the location be used instead.

Reply:
We will revisit this decision and take both designs into consideration again.